app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -106,13 +106,19 @@ def valid_origin?(uri)
   end
 
   def failure
-    error_key = params[:message].to_s.gsub(/[^\w-]/, "")
-    error_key = "generic" if error_key.blank?
+    if provider = params[:provider].presence || params[:strategy].presence
+      if authenticator = Discourse.enabled_authenticators.find { _1.name == provider }
+        provider = authenticator.display_name
+      end
+    end
+
+    if provider.blank? && Discourse.enabled_authenticators.size == 1
+      provider = Discourse.enabled_authenticators.first.display_name
+    end
 
-    flash[:error] = I18n.t(
-      "login.omniauth_error.#{error_key}",
-      default: I18n.t("login.omniauth_error.generic"),
-    ).html_safe
+    key = params[:message].to_s.gsub(/[^\w-]/, "").presence || "generic"
```

Comment:
In the case of a social login without a name, this would result in "Sorry, there was an error while trying to authenticate with generic. Please try again."? Maybe we should reword or keep a string for when the authenticator doesn't have a display name.

Reply:
> 1. there won't be any social login without a `name` since that method is mandatory

Do we need authenticators in plugins to be updated as well? Say, the apple-auth plugin? Or does that default to `name` if no `display_name` method exists? 

> the generic here is a "key"

Right, facepalm!